core/Cargo.toml

```diff
@@ -262,6 +262,7 @@ reqwest = { version = "0.11.18", features = [
 serde = { version = "1", features = ["derive"] }
 serde_json = "1"
 tokio = { version = "1.27", features = ["sync"] }
+url = "2.5.0"
```

Comment:
Please don't introduce new deps unless we have strong reasons.

Reply:
As discussed offline, we need to do url encode after base64 encode, so I add this deps, and notices that `url = "2.5.0"` is also added in somewhere else.

Reply:
> As discussed offline, we need to do url encode after base64 encode

Using `percent_encode_path` is enough, we don't need a full url encoding.

> and notices that `url = "2.5.0"` is also added in somewhere else(like [here](https://github.com/apache/opendal/blob/93e5f65bbf30df2fed4bdd95bb0685c73c6418c2/bin/ofs/Cargo.toml#L49)).

It's not part of `opendal` rust core.